### DISCUSSION TITLE
1.1 version memory question

### DISCUSSION THREAD

**Discussion by ChPi** [apache/doris#11613] [A - General / Q&A] [answered]
在1.1上向量化模式跑了select sql，Consumption只有9G，root19G，都小于进程占用的40多g。
> 在0.15版本同样的查询测试占用15g左右

感觉有的内存没释放或者没统计呢

Mem Limit: 50.21 GB
Mem Consumption: 9.35 GB
------------------------------------------------
MALLOC:    45235737016 (43140.2 MiB) Bytes in use by application
MALLOC: +            0 (    0.0 MiB) Bytes in page heap freelist
MALLOC: +    138456176 (  132.0 MiB) Bytes in central cache freelist
MALLOC: +      6161664 (    5.9 MiB) Bytes in transfer cache freelist
MALLOC: +    628776152 (  599.6 MiB) Bytes in thread cache freelists
MALLOC: +    113639424 (  108.4 MiB) Bytes in malloc metadata
MALLOC:   ------------
MALLOC: =  46122770432 (43986.1 MiB) Actual memory used (physical + swap)
MALLOC: +   1234075648 ( 1176.9 MiB) Bytes released to OS (aka unmapped)
MALLOC:   ------------
MALLOC: =  47356846080 (45163.0 MiB) Virtual address space used
MALLOC:
MALLOC:        1096226              Spans in use
MALLOC:            772              Thread heaps in use
MALLOC:           8192              Tcmalloc page size
------------------------------------------------

Id | Parent | Limit | Current Consumption | Peak Consumption | Use Count
-- | -- | -- | -- | -- | --
Root | none | none | 19G | 20G | 267
FileBlockManager | Root | none | 0 | 0 | 2
LoadChannelMgr | Root | 25G | 0 | 0 | 2
SegmentLoader | Root | 976K | 440 | 444 | 3
SegmentCache | SegmentLoader | none | 440 | 444 | 2
StoragePageCache | Root | 10G | 9G | 9G | 4
IndexPageCache | StoragePageCache | none | 11M | 11M | 2
DataPageCache | StoragePageCache | none | 9G | 9G | 2
Process | Root | 50G | 10G | 10G | 528
LastestSuccessChannelCache | Root | none | 0 | 0 | 2
FileHandlerCache | Root | none | 50K | 53K | 2
SegmentIndexCache | Root | none | 0 | 0 | 2
TabletMeta | Root | none | 4M | 4M | 2
ConvertRowset | Root | none | 0 | 0 | 2
TabletHeader | Root | none | 9M | 9M | 362
AutoCompaction | Root | none | 0 | 0 | 2

**Comment by ChPi** [comment, marked as answer]
发现PODArrayBase和BitShufflePagePreDecoder会占用不少内存

**Reply by xinyiZzz** [reply]
@ChPi  1.1.1 之前的内存统计不全，在master以及1.1.2之后的版本上实现了新的mem tracker，可以准确统计be内存使用，可以在master或1.1.2上做下同样的测试看看效果

**Reply by ChPi** [reply]
嗯，我看了 #11449 ，PODArrayBase和bitshuffle_page_page做了修改，我对c++不熟
<img width="496" alt="image" src="https://user-images.githubusercontent.com/20807051/183616434-d14cd9a9-9c95-47e0-96d8-517ec326b4d5.png">
根据这个图，是说bitshuffle_page_pre.decode分配了很多内存？如果是，不知道1.1.2版本bitshuffle_page_pre分配的内存是否也一起做了统计？

**Reply by xinyiZzz** [reply]
这是应该是 bitshuffle_page_pre.decode 这里耗时了37.9%，1.1.2上bitshuffle_page_pre这里的内存统计在了query中，可以测下试试

**Reply by ChPi** [reply]
thank you，i have found the reason and fixed it #11683
